Adding multivalue for enum field.

JaniszM opened the pull request:
Added possibility to check multiple values for one enum field if repeat parameter is set.
Query params addapted to HTTP 1.1 specification.
Some styles changed, please look at this.

sichvoge commented:
Hi @JaniszM before doing that, could you `rebase` your commits and sign our [CLA](http://www.mulesoft.org/legal/contributor-agreement.html) [here](https://api-notebook.anypoint.mulesoft.com/notebooks/#380297ed0e474010ff43)

dmartinezg commented:
Please do not modify the white-spacing rules on Gruntfile.

dmartinezg commented on dist/scripts/api-console.js:
```diff
@@ -734,27 +734,43 @@
-        $scope.isEnum = function (definition) {
-          return typeof definition['enum'] !== 'undefined';
+        $scope.isEnum = function(definition) {
+          return (typeof definition['enum'] !== 'undefined' && definition.repeat !== true);
```
```
return (typeof definition['enum'] !== 'undefined' && !!definition.repeat);
```

JaniszM replied:
Done.

dmartinezg commented on dist/scripts/api-console.js:
```diff
@@ -734,27 +734,43 @@
 
-        $scope.isBoolean = function (definition) {
+        $scope.isMultiEnum = function(definition) {
+          return (typeof definition['enum'] !== 'undefined' && definition.repeat === true);
```
return (typeof definition['enum'] !== 'undefined' && !!definition.repeat);

JaniszM replied:
Done.

dmartinezg commented:
There are a lot of syntax changes, which makes hard to understand the functional ones.

Can you help us describing what do you mean by `Query params adapted to HTTP 1.1 specification.
`?

JaniszM commented:
@sichvoge I singned CLA yesterday :) At least I think so.

@dmartinezg can you provide me formatter for your project? Currently I am using sublime + JsFormat. Then I will reformat everything agian.

For query params please refer to http://tools.ietf.org/html/rfc3986#section-3.4. Good convantion says to not use '+' in query param e.g.: instead doing 'thing=desk+lamp' use 'thing=desk&thing=lamp'.

PS: Why in grunt file do not format spaces?

dmartinezg commented:
@JaniszM, you are right, the project is missing the jshint and jscs presets, we use in other projects.

I will add them in another PR, can you please separate the syntax changes from the other ones?

Also, what effect do the CSS changes have? can you add a screenshot?

dmartinezg commented:
Can you reset tabs to 2 spaces?

We use .editorconfig (http://editorconfig.org/) to ensure consistent spacing

JaniszM commented:
Please don't merge this anyway. I found bug related with checkbox when type is boolean. I will check it and commit fix later. @dmartinezg Also I will answer to you later :)

PS: very important test is missing. There isn't any test checking checkboxes :(

dmartinezg commented:
Thanks for letting us know.

JaniszM commented:
@dmartinezg  Ok. I fixed bug. Don't bother styles now. I restored all from api-console master and added one style class to multiEnum field. Nothing else is changed. (see attached image)

Regarding spaces. I have set 2 spaces on tab (and space indent), but still can't format code like you do. Please give me some rules of formatting that I can use. These from http://editorconfig.org/ are default and you are using something more I think.

![screen shot 2015-12-04 at 15 03 01](https://cloud.githubusercontent.com/assets/14543438/11591551/e188f094-9a98-11e5-8e53-1d507690db78.png)

dmartinezg commented:
:+1: Can you please squash commits?

I'll merge it after that.

For spacing, we'll add the right jscs and jshint profiles, and I will fix the differences. We are not formatting using any linter, we format manually according to (missing) jscs rules.

JaniszM commented:
Done, commits squashed.

dmartinezg commented:
There still are 8 commits.

JaniszM commented:
Hmm, you are right. I did some mistake -.- On monday I will squash them all ;)

JaniszM commented:
Now squashed ;)

JaniszM commented:
What is status of this PR? Soon I will prepare another one.

JaniszM commented:
Hey, guys, what about this pull request? It's been a long time since everything is ready to merge ;)

dmartinezg commented:
Hi @JaniszM, as I said before, this is a beast of a PR, because of all the styling changes, and on top of it, they are not consistent with current style.

Please revert the spacing changes, and leave the PR to a minimum of changes.

@xaka, does this have any impact on the next changes for 1.0?

JaniszM commented:
Huh, @dmartinezg as I ask before, would be nice to have some formatter rules for this ;) Reverting this stuff can be time consuming and... manually this not have sense :x Please give me some rules, at least formal, and I will try to set up my formatter.

JaniszM commented:
Ok, done, I think I reverted all spacing changes. Is it fine now?

JaniszM commented:
Oh My God. I even don't remember what It was for -.- If I find some time I will check it again :o It was so long ago...

sichvoge commented:
Indeed its been a while :D Sorry for the late response.

jarrodek commented:
Hi, thank you for the PR. It relates to older version of the console which is no longer maintained.

JaniszM commented:
@jarrodek At that time it was up to date ;D

jarrodek commented:
@JaniszM yes, it was :( I am sorry for that.

JaniszM commented:
No problem, don't worry.
